samples/net/sntp/src/main.c

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (c) 2024 Embeint Inc
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+ #include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(net_sntp_sample, LOG_LEVEL_DBG);
+
+#include <time.h>
+
+#include <zephyr/net/socket.h>
+#include <zephyr/net/socket_service.h>
+#include <zephyr/net/sntp.h>
+#include <zephyr/posix/arpa/inet.h>
+
+#include "net_sample_common.h"
+
+static K_SEM_DEFINE(sntp_async_received, 0, 1);
+static void sntp_service_handler(struct net_socket_service_event *pev);
+
+NET_SOCKET_SERVICE_SYNC_DEFINE_STATIC(service_sntp_async, sntp_service_handler, 1);
+
+int dns_query(const char *host, uint16_t port, int family, int socktype, struct sockaddr *addr,
+	      socklen_t *addrlen)
+{
+	struct zsock_addrinfo hints = {
+		.ai_family = family,
+		.ai_socktype = socktype,
+	};
+	struct zsock_addrinfo *res = NULL;
+	char addr_str[INET6_ADDRSTRLEN] = {0};
+	int rc;
+
+	/* Perform DNS query */
+	rc = zsock_getaddrinfo(host, NULL, &hints, &res);
+	if (rc < 0) {
+		LOG_ERR("zsock_getaddrinfo failed (%d, errno %d)", rc, errno);
+		return rc;
+	}
+	/* Store the first result */
+	*addr = *res->ai_addr;
+	*addrlen = res->ai_addrlen;
+	/* Free the allocated memory */
+	zsock_freeaddrinfo(res);
+
+	if (addr->sa_family == AF_INET) {
+		struct sockaddr_in *ipv4 = (struct sockaddr_in *)addr;
+
+		ipv4->sin_port = htons(port);
+		inet_ntop(ipv4->sin_family, &ipv4->sin_addr, addr_str, sizeof(addr_str));
+	} else {
+		struct sockaddr_in6 *ipv6 = (struct sockaddr_in6 *)addr;
+
+		ipv6->sin6_port = htons(port);
+		inet_ntop(ipv6->sin6_family, &ipv6->sin6_addr, addr_str, sizeof(addr_str));
+	}
```

Comment:
I think we could replace this by so no need to check the family here.
```
net_sin(addr)->sin_port = htons(port);
zsock_inet_ntop(addr->sa_family, &net_sin(addr)->sin_addr, addr_str, sizeof(addr_str));
```
This works because the port number and address are pointing to the same place both in `sockaddr_in` and `sockaddr_in6`
Also there is a zsock variant for inet_ntop so that you do not need to include the `arpa/inet.h`